### PR TITLE
feat: SfRadio details component added

### LINKED ISSUE #845: [FEATURE] Toggle content in SfRadio
**Describe how this feature will help us improve**
Currently, we have SfRadio with the label, we need to improve it by adding additional content in dropdown according to new designs.
~~- add a slot for price~~
- add a slot for subtitle

image

**Additional context**
https://www.figma.com/file/N0Ct95cSAoODNv7zYS01ng/Storefront-UI-Design-System?node-id=216%3A4

### PR DESCRIPTION
**Pull request by AdamPawlinski**
# Related issue
Closes #845

# Scope of work
SfRadio, details component added 

# Screenshots of visual changes

image

# Checklist

- [x] No commented blocks of code left
- [ ] Run tests and docs
- [x] Self code-reviewed

### CODE REVIEW

File: packages/shared/styles/components/molecules/SfRadio.scss [hunk 1]
```diff
@@ -56,6 +56,17 @@
     flex: 1;
     margin: var(--radio-content-margin, 0 0 0 var(--spacer-sm));
   }
+  &__details {
+    margin: var(--radio-details-margin, var(--spacer-2xs) 0 0 0);
+    color: var(--radio-details-color, var(--c-text));
```
Comment (pspaczek) [inline packages/shared/styles/components/molecules/SfRadio.scss:61]: @AdamPawlinski adjust color to design

File: packages/shared/styles/components/molecules/SfRadio.scss [hunk 2]
```diff
@@ -56,6 +56,17 @@
     flex: 1;
     margin: var(--radio-content-margin, 0 0 0 var(--spacer-sm));
   }
+  &__details {
+    margin: var(--radio-details-margin, var(--spacer-2xs) 0 0 0);
```
Comment (pspaczek) [inline packages/shared/styles/components/molecules/SfRadio.scss:60]:
```suggestion
    margin: var(--radio-details-margin, var(--spacer-xs) 0 0 0);
```
8px looks better than 4px

**Comment by pspaczek** [comment]
@AdamPawlinski looks good for me, please check my comment and adjust CSS to desing

**Comment by pspaczek** [comment]
@martaradziszewska please check this PR, I need a second reviewer here

File: packages/shared/styles/components/molecules/SfRadio.scss (hunk 1 above)
Comment (AdamPawlinski) [reply packages/shared/styles/components/molecules/SfRadio.scss:61]: ok, I overlooked that

File: packages/shared/styles/components/molecules/SfRadio.scss (hunk 2 above)
Comment (AdamPawlinski) [reply packages/shared/styles/components/molecules/SfRadio.scss]: changed

**Comment by martaradziszewska** [comment]
@pspaczek @AdamPawlinski looks great! :) To show more possibilities I would add a pure story of SfRadio with the label only, wdyt?  :)

**Comment by AdamPawlinski** [comment]
> @pspaczek @AdamPawlinski looks great! :) To show more possibilities I would add a pure story of SfRadio with the label only, wdyt? :)

@martaradziszewska  I'm not sure if it is needed you can just delete details and description.

**Comment by martaradziszewska** [comment]
@AdamPawlinski yes, but someone won't know about it that it's easily removable :) There are some cases with simple radio input, but we are showing only an advanced version of it. @pspaczek @justyna-13 wdyt?

**Comment by pspaczek** [comment]
@martaradziszewska now it's ok. Common stories have all options and slot props present other options 👍 IMO it's ok.

**Pull request merged**